daemon/user/safelauncher.go

```diff
@@ -0,0 +1,98 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package user
+
+import (
+	"net/url"
+	"os/exec"
+
+	"fmt"
+
+	"github.com/godbus/dbus"
+)
+
+const (
+	safeLauncherIntrospectionXML = `
+	<interface name='com.canonical.SafeLauncher'>
+		<method name='OpenURL'>
+			<arg type='s' name='url' direction='in'/>
+		</method>
+	</interface>`
+)
+
+var (
+	allowedUrlSchemes = []string{"http", "https", "mailto"}
```

Comment:
in the mimetype handler of the .desktop file we install inside core: https://github.com/snapcore/core/blob/master/live-build/hooks/500-create-xdg-wrapper.binary#L23
and in the mimeapps.list we implement:
https://github.com/snapcore/core/blob/master/live-build/hooks/500-create-xdg-wrapper.binary#L28
protocols not defined in there will simply be ignored.